### PR TITLE
PowerSNO: Create CI on PowerVS

### PR DESCRIPTION
**Pull request by hamzy**
We are creating a CI for 4.14 to test Single Node OpenShift jobs.

### CODE REVIEW

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by jaypoulz** [comment]
A few notes - we'll definitely want a rehearsal job run of the single-node test.
It would be good to schedule a rehearsal for the regular power vs test as well, just to ensure that none of the changes broke that.

Next, I would recommend that maybe we have a 3-1 split for jobs. That is to say, 3 times a day, the e2e is run against the full 5-6 node cluster, and once a day it's run against the single node configuration.

**Comment by jaypoulz** [comment]
After relooking at the configs, it looks like you already did this. :)
> Next, I would recommend that maybe we have a 3-1 split for jobs. That is to say, 3 times a day, the e2e is run against the full 5-6 node cluster, and once a day it's run against the single node configuration. (This is based on the observation that it appears we run configurations for the full 5-6 node cluster 4 times per day currently.

**Comment by hamzy** [comment]
> It would be good to schedule a rehearsal for the regular power vs test as well, just to ensure that none of the changes broke that.

Just did and it looks like I didn't :) 

```
CONTROL_PLANE_REPLICAS=3
WORKER_REPLICAS=2
Collecting pyyaml
  Downloading https://files.pythonhosted.org/packages/cd/e5/af35f7ea75cf72f2cd079c95ee16797de7cd71f29ea7c68ae5ce7be1eda0/PyYAML-6.0.1.tar.gz (125kB)
Installing collected packages: pyyaml
  Running setup.py install for pyyaml: started
    Running setup.py install for pyyaml: finished with status 'done'
Successfully installed pyyaml-6.0.1
PLATFORM_ARGS_COMPUTE=
PLATFORM_ARGS_WORKER=
CONFIG_PLATFORM_COMPUTE=  platform:
    powervs: {}
CONFIG_PLATFORM_WORKER=  platform:
    powervs: {}
```

**Comment by jaypoulz** [comment]
Wierd because we can now see a rehearsal for the 4.13 e2e run. What I don't see is a rehearsal for the single node e2e run, but maybe we're waiting for this current test to finish?

**Comment by hamzy** [comment]
> Wierd because we can now see a rehearsal for the 4.13 e2e run. What I don't see is a rehearsal for the single node e2e run, but maybe we're waiting for this current test to finish?

There has to be a bug because I've rehearsed it like three times already and watched it progress via the command line.  Unfortunately, I keep running into the frustratingly common master nodes not getting their ignition file networking error. :( I'm thinking of pushing a temporary change and test the -2 lease instead of the -1 lease.  At least to watch a cluster deploy.

**Comment by hamzy** [comment]
Here are the artifacts for four attempts:

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/41458/rehearse-41458-periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs/

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

**Review by jaypoulz — APPROVED** [review]
This looks good to me. My only guidance is that I'd like to see a rehearsal run before this merges and that the edits to the single node worker refs to modify the timing behavior looks like it may no longer be needed for this patch, so you may want to remove it to save yourself extra reviews.

**Comment by hamzy** [comment]
Hello single-node approvers @romfreiman @eranco74 @tsorya @osherdp @omertuc @adriengentil
Could one of you review and approve the changes?

**Comment by hamzy** [comment]
I still think the timing bumps are needed in the PowerVS world because sadly the nodes take a long time to come online.

**Comment by jaypoulz** [comment]
What I'm saying is that I don't see any references to tests where we actually call into add-workers. I think when we last discussed this patch, we determined that we didn't want to cover that test workflow. we just want a basic smoke test for single-node running as a single node.

**Comment by hamzy** [comment]
@jaypoulz Yeah, I had switched from the openshift-e2e-powervs-single-node-workers workflow to the openshift-e2e-powervs-single-node workflow which does not add workers.  I will remove the add-worker cruft.

**Comment by hamzy** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-single-node-ppc64le-powervs

File: ci-operator/step-registry/single-node/e2e/test/single-node-e2e-test-commands.sh
Comment (omertuc) [reply ci-operator/step-registry/single-node/e2e/test/single-node-e2e-test-commands.sh:155]: The only difference between those two files should be `--max-parallel-tests 15 \`

Comment (omertuc) [reply ci-operator/step-registry/single-node/e2e/test/single-node-e2e-test-commands.sh:155]: https://github.com/openshift/release/pull/43704

Comment (omertuc) [reply ci-operator/step-registry/single-node/e2e/test/single-node-e2e-test-commands.sh:155]: Ideally the max parallel tests should be configurable on the original file but I got pushback when asking to make it configurable

Comment (omertuc) [reply ci-operator/step-registry/single-node/e2e/test/single-node-e2e-test-commands.sh:155]: #43704 merged, so please add this to the original file as well

Comment (hamzy) [reply ci-operator/step-registry/single-node/e2e/test/single-node-e2e-test-commands.sh:168]: @omertuc 
I just rebased and noticed that my small change to this file remains the same.  It will cleanly apply to your new changes.  Does this satisfy your concern?

Comment (hamzy) [reply ci-operator/step-registry/single-node/e2e/test/single-node-e2e-test-commands.sh:168]: Or do you want me to apply my changes "upstream" to that other file?

Comment (omertuc) [reply ci-operator/step-registry/single-node/e2e/test/single-node-e2e-test-commands.sh:168]: >Or do you want me to apply my changes "upstream" to that other file?

This

Comment (hamzy) [reply ci-operator/step-registry/single-node/e2e/test/single-node-e2e-test-commands.sh:168]: In that case, wouldn't I have to change both files then?

Comment (omertuc) [reply ci-operator/step-registry/single-node/e2e/test/single-node-e2e-test-commands.sh:168]: Yes

Comment (hamzy) [reply ci-operator/step-registry/single-node/e2e/test/single-node-e2e-test-commands.sh:168]: Done!

**Comment by hamzy** [comment]
Hello openshift-approvers:
@deads2k @dgoodwin @stbenjam @vrutkovs @wking

Would one of you be willing to review this PR for me please?

**Pull request merged**
